Algorithm/QCAlgorithm.Trading.cs

```diff
@@ -957,48 +957,40 @@ public decimal CalculateOrderQuantity(Symbol symbol, decimal target)
             decimal marginRequired;
             decimal orderValue;
             decimal orderFees;
-            var feeToPriceRatio = 0;
+            var feeToPriceRatio = 0m;
 
             // compute the initial order quantity
-            decimal orderQuantity = targetOrderValue / unitPrice;
+            var orderQuantity = targetOrderValue / unitPrice;
 
-            if (orderQuantity % security.SymbolProperties.LotSize != 0)
-            {
-                orderQuantity = orderQuantity - (orderQuantity % security.SymbolProperties.LotSize);
-            }
-
-            var iterations = 0;
+            // rounding off Order Quantity to the nearest multiple of Lot Size
+            orderQuantity -= orderQuantity % security.SymbolProperties.LotSize;
 
             do
             {
-                // decrease the order quantity
-                if (iterations > 0)
-                {
-                    // if fees are high relative to price, we reduce the order quantity faster
-                    if (feeToPriceRatio > 0)
-                        orderQuantity -= feeToPriceRatio;
-                    else
-                        orderQuantity--;
-                }
+                // reduce order quantity by feeToPriceRatio, since it is faster than by lot size
+                // if it becomes nonpositive, return zero
+                orderQuantity -= feeToPriceRatio;
+                if (orderQuantity <= 0) return 0;
 
                 // generate the order
                 var order = new MarketOrder(security.Symbol, orderQuantity, UtcTime);
                 orderValue = order.GetValue(security);
                 orderFees = security.FeeModel.GetOrderFee(security, order);
-                feeToPriceRatio = (int)(orderFees / unitPrice);
+
+                // calculate feeToPriceRatio, if lower than lot size, use lot size instead
+                feeToPriceRatio = orderFees / unitPrice;
+                if (feeToPriceRatio < security.SymbolProperties.LotSize)
+                {
+                    feeToPriceRatio = security.SymbolProperties.LotSize;
+                }
 
                 // calculate the margin required for the order
                 marginRequired = security.MarginModel.GetInitialMarginRequiredForOrder(security, order);
 
-                iterations++;
+            } while (marginRequired > marginRemaining || orderValue + orderFees > targetOrderValue);
 
-            } while (orderQuantity > 0 && (marginRequired > marginRemaining || orderValue + orderFees > targetOrderValue));
-
-            //Rounding off Order Quantity to the nearest multiple of Lot Size
-            if (orderQuantity % security.SymbolProperties.LotSize != 0)
-            {
-                orderQuantity = orderQuantity - (orderQuantity % security.SymbolProperties.LotSize);
-            }
+            var remainder = orderQuantity % security.SymbolProperties.LotSize;
+            if (remainder > 0) orderQuantity -= remainder;
```

Comment:
Not important, but this could easily be written without an if statement:

```csharp
orderQuantity -= orderQuantity % security.SymbolProperties.LotSize;
```